miraclelinux/meta-debian
# flac: Update version from 1.3.2-3+deb10u2 to 1.3.2-3+deb10u3

# Purpose of pull request
This version fixes following CVEs.
- CVE-2020-22219

DLA: DLA 3581-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build and run the qemuarm64 image
2. Download sample file
3. Run commands to check the flac working

### Build and run the qemuarm64 image
Add the following to local.conf.
```
IMAGE_INSTALL_append = " flac"
DISTRO_FEATURES_append = " pam"
IMAGE_INSTALL_append = " openssh"
IMAGE_ROOTFS_EXTRA_SPACE = "1048576"
```
```
$ bitbake core-image-minimal
$ runqemu qemuarm64 nographic
```
### Download sample file
Download the sample1.wav file from the "https://filesamples.com/formats/wav" site and transfer it to the QEMU environment using scp.

### Run commands to check the flac working
```
# ls -l
# flac -V ./sample1.wav -o sample1.w64
# flac -V --decode ./sample1.w64 -o sample1-1.wav
# ls -l
# file *
```

## Test result
### Build package
Build succeeded.
```
$ bitbake flac
Parsing recipes: 100% |#################################################################################################################| Time: 0:01:37
Parsing of 1362 .bb files complete (0 cached, 1362 parsed). 2382 targets, 81 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-20.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.8"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-flac_1.3.2-3+deb10u3:37685b35e561c1647479cd644e9f4ee3aa23059b"
meta-debian-extended = "HEAD:1faec1396bc9871f46202287446757993c392468"
meta-emlinux         = "HEAD:4dd2259ccc4eaaa6dece7f164eb2827bd09eadc4"
meta-emlinux-private = "HEAD:8207525289999d2c41983eb8155127b62621cdb1"

NOTE: Fetching uninative binary shim from http://downloads.yoctoproject.org/releases/uninative/2.9/x86_64-nativesdk-libc.tar.xz;sha256sum=d07916b95c419c81541a19c8ef0ed8cbd78ae18437ff28a4c8a60ef40518e423
Initialising tasks: 100% |##############################################################################################################| Time: 0:00:01
Sstate summary: Wanted 86 Found 0 Missed 86 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 614 tasks of which 0 didn't need to be rerun and all succeeded.
```

### Package test
Encoding/decoding with flac command is working.
And all converted files could be played on Ubuntu20.04 host PC.
```
root@qemuarm64:~# ls -l
total 21036
-rw-r--r-- 1 root root 21537382 Oct 12 04:22 sample1.wav
root@qemuarm64:~# flac -V ./sample1.wav -o sample1.w64

flac 1.3.2
Copyright (C) 2000-2009  Josh Coalson, 2011-2016  Xiph.Org Foundation
flac comes with ABSOLUTELY NO WARRANTY.  This is free software, and you are
welcome to redistribute it under certain conditions.  Type `flac' for details.

sample1.wav: WARNING: skipping unknown chunk 'LIST' (use --keep-foreign-metadata to keep)
sample1.wav: Verify OK, wrote 12621556 bytes, ratio=0.586
root@qemuarm64:~# flac -V --decode ./sample1.w64 -o sample1-1.wav

flac 1.3.2
Copyright (C) 2000-2009  Josh Coalson, 2011-2016  Xiph.Org Foundation
flac comes with ABSOLUTELY NO WARRANTY.  This is free software, and you are
welcome to redistribute it under certain conditions.  Type `flac' for details.

sample1.w64: done         
root@qemuarm64:~# ls -l
total 54400
-rw-r--r-- 1 root root 21537348 Oct 12 04:22 sample1-1.wav
-rw-r--r-- 1 root root 12621556 Oct 12 04:22 sample1.w64
-rw-r--r-- 1 root root 21537382 Oct 12 04:22 sample1.wav
root@qemuarm64:~# file *
sample1-1.wav: RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, stereo 44100 Hz
sample1.w64:   FLAC audio bitstream data, 16 bit, stereo, 44.1 kHz, 5384326 samples
sample1.wav:   RIFF (little-endian) data, WAVE audio, Microsoft PCM, 16 bit, stereo 44100 Hz
```



All converted files could be played on Ubuntu20.04 host PC.
